core/domain/question_services.py

```diff
@@ -131,7 +139,8 @@ def get_questions_by_ids(question_ids):
     question_model_list = question_models.QuestionModel.get_multi(question_ids)
     questions = []
     for question_model in question_model_list:
-        questions.append(get_question_from_model(question_model))
+        if question_model is not None:
+            questions.append(get_question_from_model(question_model))
```

Comment:
This seems wrong. You may have a return list with different numbers of elements from the input list. Did you include backend tests for the case where one or more question IDs in the list could not be found?

Reply:
Done, added None for such elements and added tests.